handler/error.go

```diff
@@ -0,0 +1,43 @@
+package handler
+
+// Error is the packages error type
+type Error struct{
+	err        error
+	message    string
+	statusCode int
+	logData    map[string]interface{}
+}
+
+// Error satisfies the standard library Go error interface
+func (e Error) Error() string{
+	if e.err == nil{
+		return "nil"
+	}
+	return e.err.Error()
+}
+
+// Unwrap implements the standard library Go unwrapper interface
+func (e Error) Unwrap() error{
+	return e.err
+}
+
+// Code satisfies the coder interface which is used to recover a
+// HTTP status code from an error
+func (e Error) Code() int{
+	return e.statusCode
+}
+
+// Message satisfies the messanger interface which is used to specify
```

Comment:
typo here too:
```suggestion
// Message satisfies the messager interface which is used to specify
```
I think it might be nicer to name it the messenger interface though.